packages/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -365,20 +395,32 @@ class DescopeWc extends BaseDescopeWc {
       htmlUrl: getContentUrl(projectId, `${readyScreenId}.html`),
       htmlLocaleUrl:
         filenameWithLocale && getContentUrl(projectId, filenameWithLocale),
+      samlIdpUsername,
     };
 
     const lastAuth = getLastAuth(loginId);
 
     // If there is a start screen id, next action should start the flow
-    // But if oidcIdpStateId is not empty, this optimization doesn't happen
-    if (showFirstScreenOnExecutionInit(startScreenId, oidcIdpStateId)) {
+    // But if oidcIdpStateId, samlIdpStateId, ssoAppId is not empty, this optimization doesn't happen
+    if (
+      showFirstScreenOnExecutionInit(
```

Comment:
I see that we are calculating this condition twice, 
can we store it in a variable instead of calling it twice?

Reply:
I think this double calculation is required as by doing a quick test the optimization you suggest causes tests to fail, so I dont wanna go into debug all this in this pr.